scheduler/servicestatemanager/servicestatemanager.go

```diff
@@ -73,6 +73,7 @@ type ServiceStateChangeBatch struct {
 
 // CancellableService is a service whose scheduling may be canceled by a channel
 type CancellableService struct {
+	sync.Mutex
```

Comment:
These objects get passed around a lot, and aren't always cancelled from a BatchServiceStateManager member method, so the alternative would be to have a package-level lock, which makes me nervous about potential deadlock.